Assets/Scripts/TrainingArena.cs

```diff
@@ -251,16 +253,21 @@ private void ApplyNewArenaConfiguration(ArenaConfiguration newConfiguration)
         }
 
         _arenaConfiguration = newConfiguration;
+        Debug.Log($"Setting passMark to: {newConfiguration.passMark}");
+        // Ensure passMark is set correctly
+        _arenaConfiguration.passMark = newConfiguration.passMark;
```

Comment:
Why does `passMark` need special treatment? Could anything else be broken in the same way `passMark` was?

Reply:
There isn't a special treatment but its an important parameter:

1. It defines the _threshold_ for agents passing to the next arena/episode.
2. It's relied on by _showNotification_ feature (the success or failure gradients are shown according to passMark value )
3. It's a core parameter (along with _timeLimit_) for each arena configuration in the yaml.

Reply:
In other words: my understanding of what L255 `_arenaConfiguration = newConfiguration` would be doing is that for any field `x`,  `_arenaConfiguration.x` will be equal to `newConfiguration.x`. Is there something special about `passMark` that means this isn't happening, or am I misunderstanding what the result of L255 is?